src/Application/QueryStringParser.php

```diff
@@ -38,11 +43,33 @@ private function splitQueryString( string $queryString ): array {
 		);
 	}
 
+	private function isInstancePart( string $part ): bool {
+		return str_starts_with( $part, 'haswbstatement:' );
+	}
+
 	private function isFacetPart( string $part ): bool {
 		return str_starts_with( $part, 'haswbfacet:' )
 			|| str_starts_with( $part, '-haswbfacet:' );
 	}
 
+	/**
+	 * @param array<string, NumericPropertyId|ItemId> $instance
+	 * @return array<string, NumericPropertyId|ItemId>
+	 */
+	private function handleInstancePart( string $part, array &$instance ): array {
```

Comment:
`$instance` should not be by-reference

Name isn't great. What about `itemTypes`?

Reply:
Parameter and return types should change to `ItemId[]`

Reply:
Btw, the existing return type confused me for a while. First I wondered "why is this an array" and then I thought "because an item can have multiple types". But no, it's an array because you return two different things.

I suppose it is fine for the MVP to only look at the first item type. This will work for the first use case of the extension.